From ad1c07155f937f4a7be84a4249158ee9b6d263a6 Mon Sep 17 00:00:00 2001
From: Julien Balestra <julien.balestra@blablacar.com>
Date: Fri, 6 Jan 2017 15:55:35 +0100
Subject: [PATCH] Syslog

---
 pkg/kubelet/rkt/rkt.go | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/pkg/kubelet/rkt/rkt.go b/pkg/kubelet/rkt/rkt.go
index ce797fe..7ff0ab3 100644
--- a/pkg/kubelet/rkt/rkt.go
+++ b/pkg/kubelet/rkt/rkt.go
@@ -1111,6 +1111,15 @@ func (r *Runtime) getSelinuxContext(opt *v1.SELinuxOptions) (string, error) {
 	return strings.Join(ctx, ":"), nil
 }
 
+func getPrefixPodName(podName string) string {
+	n := strings.Split(podName, "-")
+	if len(n) > 1 && len(n[len(n) - 1]) == 5 {
+		return strings.Join(n[:len(n) - 1], "-")
+	} else {
+		return strings.Join(n[:], "-")
+	}
+}
+
 // preparePod will:
 //
 // 1. Invoke 'rkt prepare' to prepare the pod, and get the rkt pod uuid.
@@ -1172,6 +1181,8 @@ func (r *Runtime) preparePod(pod *v1.Pod, podIP string, pullSecrets []v1.Secret,
 
 	hostNetwork := kubecontainer.IsHostNetworkPod(pod)
 	units := []*unit.UnitOption{
+		newUnitOption("Unit", "Description", pod.Namespace + "." + pod.GenerateName),
+		newUnitOption("Service", "SyslogIdentifier", pod.Namespace + "." + getPrefixPodName(pod.Name)),
 		newUnitOption("Service", "ExecStart", runPrepared),
 		newUnitOption("Service", "ExecStopPost", markPodFinished),
 		// This enables graceful stop.
-- 
2.7.4

