From d35a4fdf823216b279a139db46c307c3ff9d08da Mon Sep 17 00:00:00 2001
From: JulienBalestra <julien.balestra@gmail.com>
Date: Fri, 13 Oct 2017 17:37:18 +0200
Subject: [PATCH] debug the app sandbox

---
 rktlet/runtime/helpers.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rktlet/runtime/helpers.go b/rktlet/runtime/helpers.go
index ca692cb..545bf24 100644
--- a/rktlet/runtime/helpers.go
+++ b/rktlet/runtime/helpers.go
@@ -400,7 +400,7 @@ func maybeCreateHostPathVolume(mount *runtimeApi.Mount) (created bool, err error
 }
 
 func generateAppSandboxCommand(req *runtimeApi.RunPodSandboxRequest, uuidfile, stage1Name, networkPluginName string) ([]string, error) {
-	cmd := []string{"app", "sandbox", "--uuid-file-save=" + uuidfile}
+	cmd := []string{"app", "--debug", "sandbox", "--uuid-file-save=" + uuidfile}
 
 	// annotation takes preference over configuration
 	if val, ok := req.Config.Annotations[k8sRktStage1NameAnno]; ok {
-- 
2.7.4

