From 37cbd0baddf0d3e9f58b0854b5553a0e3dcea2f1 Mon Sep 17 00:00:00 2001
From: JulienBalestra <julien.balestra@gmail.com>
Date: Thu, 21 Sep 2017 18:18:32 +0200
Subject: [PATCH] Insecure the app sandbox

---
 rktlet/runtime/helpers.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rktlet/runtime/helpers.go b/rktlet/runtime/helpers.go
index 528c2f2..0b87c1a 100644
--- a/rktlet/runtime/helpers.go
+++ b/rktlet/runtime/helpers.go
@@ -343,7 +343,7 @@ func generateAppSandboxCommand(req *runtimeApi.RunPodSandboxRequest, uuidfile, s
 		}
 	}
 
-	cmd = append(cmd, "--insecure-options=seccomp")
+	cmd = append(cmd, "--insecure-options=all")
 
 	if sc := req.GetConfig().GetLinux().GetSecurityContext(); sc != nil {
 		if sc.Privileged {
-- 
2.7.4

