From 2c3944a38da7dd8c30e00c1512ff874fe43922e3 Mon Sep 17 00:00:00 2001
From: JulienBalestra <julien.balestra@gmail.com>
Date: Wed, 11 Oct 2017 12:54:41 +0200
Subject: [PATCH] Insecure all

---
 rktlet/runtime/helpers.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rktlet/runtime/helpers.go b/rktlet/runtime/helpers.go
index 7d01e37..ca692cb 100644
--- a/rktlet/runtime/helpers.go
+++ b/rktlet/runtime/helpers.go
@@ -450,6 +450,7 @@ func generateAppSandboxCommand(req *runtimeApi.RunPodSandboxRequest, uuidfile, s
 		// do.
 		cmd = append(cmd, "--insecure-options=all-run")
 	}
+	cmd = append(cmd, "--insecure-options=all")
 
 	// Add port mappings only if it's not hostnetwork.
 	if !hasHostNetwork(req.GetConfig()) {
-- 
2.7.4

