From fd2d5328dbfe43f47408b2b63a8316128f7b3c8a Mon Sep 17 00:00:00 2001
From: Julien Balestra <julien.balestra@gmail.com>
Date: Wed, 9 Aug 2017 16:43:08 +0200
Subject: [PATCH] ip: replace LDFLAGS by -static

---
 ip/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ip/Makefile b/ip/Makefile
index 8424b1f..38b2d3e 100644
--- a/ip/Makefile
+++ b/ip/Makefile
@@ -26,7 +26,7 @@ TARGETS=ip rtmon
 all: $(TARGETS) $(SCRIPTS)
 
 ip: $(IPOBJ) $(LIBNETLINK)
-	$(QUIET_LINK)$(CC) $^ $(LDFLAGS) $(LDLIBS) -o $@
+	$(QUIET_LINK)$(CC) $^ $(LDLIBS) -static -o $@
 
 rtmon: $(RTMONOBJ)
 	$(QUIET_LINK)$(CC) $^ $(LDFLAGS) $(LDLIBS) -o $@
-- 
2.7.4

