From 7c53c4cb8fbb1521705061e2fb10cab5e976bcfb Mon Sep 17 00:00:00 2001
From: Julien Balestra <julien.balestra@gmail.com>
Date: Thu, 16 Mar 2017 14:48:34 +0100
Subject: [PATCH] Kubelet:rkt Create any missing hostPath Volumes

---
 pkg/kubelet/rkt/rkt.go | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/pkg/kubelet/rkt/rkt.go b/pkg/kubelet/rkt/rkt.go
index 91b5dfa..37ac3d1 100644
--- a/pkg/kubelet/rkt/rkt.go
+++ b/pkg/kubelet/rkt/rkt.go
@@ -1311,6 +1311,20 @@ func (r *Runtime) setupPodNetwork(pod *api.Pod) (string, string, error) {
 	return netnsName, status.IP.String(), nil
 }
 
+// For a hostPath volume: rkt doesn't create any missing volume on the node/host so we need to create it
+func createHostPathVolumes(pod *v1.Pod) (err error) {
+	for _, v := range pod.Spec.Volumes {
+		if v.VolumeSource.HostPath != nil {
+			err = os.MkdirAll(v.HostPath.Path, os.ModePerm)
+			if err != nil && os.IsNotExist(err) {
+				return err
+			}
+			glog.V(4).Infof("Created volume HostPath %q for Pod %q", v.HostPath.Path, format.Pod(pod))
+		}
+	}
+	return nil
+}
+
 // RunPod first creates the unit file for a pod, and then
 // starts the unit over d-bus.
 func (r *Runtime) RunPod(pod *v1.Pod, pullSecrets []v1.Secret) error {
@@ -1319,6 +1333,12 @@ func (r *Runtime) RunPod(pod *v1.Pod, pullSecrets []v1.Secret) error {
 	var err error
 	var netnsName string
 	var podIP string
+
+	err = createHostPathVolumes(pod)
+	if err != nil {
+		return err
+	}
+
 	netnsName, podIP, err = r.setupPodNetwork(pod)
 	if err != nil {
 		r.cleanupPodNetwork(pod)
-- 
2.7.4

