From 37ccc917ced210baf1cc2ad613c302fdfa16b954 Mon Sep 17 00:00:00 2001
From: JulienBalestra <julien.balestra@gmail.com>
Date: Sat, 15 Apr 2017 12:04:16 +0200
Subject: [PATCH] Kubelet:rkt Fix the hostPath Volume creation

---
 pkg/kubelet/rkt/rkt.go | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/pkg/kubelet/rkt/rkt.go b/pkg/kubelet/rkt/rkt.go
index 2d9817f..b2eeca0 100644
--- a/pkg/kubelet/rkt/rkt.go
+++ b/pkg/kubelet/rkt/rkt.go
@@ -1315,6 +1315,23 @@ func (r *Runtime) setupPodNetwork(pod *v1.Pod) (string, string, error) {
 	return netnsName, status.IP.String(), nil
 }
 
+// For a hostPath volume: rkt doesn't create any missing volume on the node/host so we need to create it
+func createHostPathVolumes(pod *v1.Pod) (err error) {
+	for _, v := range pod.Spec.Volumes {
+		if v.VolumeSource.HostPath != nil {
+			_, err = os.Stat(v.HostPath.Path)
+			if os.IsNotExist(err) {
+				if err = os.MkdirAll(v.HostPath.Path, os.ModePerm); err != nil {
+					glog.Errorf("Create volume HostPath %q for Pod %q failed: %q", v.HostPath.Path, format.Pod(pod), err.Error())
+					return err
+				}
+				glog.V(4).Infof("Created volume HostPath %q for Pod %q", v.HostPath.Path, format.Pod(pod))
+			}
+		}
+	}
+	return nil
+}
+
 // RunPod first creates the unit file for a pod, and then
 // starts the unit over d-bus.
 func (r *Runtime) RunPod(pod *v1.Pod, pullSecrets []v1.Secret) error {
@@ -1323,6 +1340,12 @@ func (r *Runtime) RunPod(pod *v1.Pod, pullSecrets []v1.Secret) error {
 	var err error
 	var netnsName string
 	var podIP string
+
+	err = createHostPathVolumes(pod)
+	if err != nil {
+		return err
+	}
+
 	netnsName, podIP, err = r.setupPodNetwork(pod)
 	if err != nil {
 		r.cleanupPodNetwork(pod)
-- 
2.7.4

